### PR TITLE
Add `Contract` functions to load CBOR from hex

### LINKED ISSUE #850: Add a function to load scripts from hex, without envelope
`textEnvelopeBytes` is useful, but if the user has just cborhex, it's tedious to wrap it.

### PR DESCRIPTION
**Pull request by MitchyCola**
Closes #850 .

### Pre-review checklist

- [x] All code has been formatted using our config (`make format`)
- [x] Any new API features or modification of existing behavior is covered as defined in our [test plan](https://github.com/Plutonomicon/cardano-transaction-lib/blob/develop/doc/test-plan.md)
- [x] The changelog has been updated under the `## Unreleased` header, using the appropriate sub-headings (`### Added`, `### Removed`, `### Fixed`), and the links to the appropriate issues/PRs have been included

### CODE REVIEW

File: src/Contract/Cbor.purs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+import Contract.Monad (Contract)
+import Data.Newtype (wrap)
+import Effect.Exception (error)
+import Types.ByteArray (ByteArray)
+import Types.ByteArray (hexToByteArray) as BA
+import Types.Cbor (Cbor)
+import Types.CborBytes (CborBytes)
+
+hexToByteArray :: String -> Contract () ByteArray
```
Comment (klntsky) [inline src/Contract/Cbor.purs:17]: There is no reason for this function to be impure.

**Review by klntsky — CHANGES_REQUESTED** [review]
.

File: src/Contract/Cbor.purs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+module Contract.Cbor
+  ( module ByteArray
+  , module CborBytes
+  , hexToCbor
+  ) where
+
+import Contract.Prelude
+
+import Ctl.Internal.Types.ByteArray (hexToByteArray) as ByteArray
```
Comment (klntsky) [inline src/Contract/Cbor.purs:9]: `hexToByteArray` is exported from Contract.Prim.ByteArray, no need to re-export

**Review by klntsky — CHANGES_REQUESTED** [review]

File: src/Contract/Cbor.purs (hunk 2 above)
Comment (MitchyCola) [reply src/Contract/Cbor.purs]: Fixed

File: src/Contract/Cbor.purs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  ) where
+
+import Contract.Prelude
+
+import Ctl.Internal.Types.Cbor (Cbor)
+import Ctl.Internal.Types.CborBytes (hexToCborBytes) as CborBytes
+import Data.Newtype (wrap)
+
+-- | Decode a hexadecimal string into a `Cbor`.
+-- | Results in `Nothing` if the string contains a non-hexadeciaml number,
```
Comment (Luis-omega) [inline src/Contract/Cbor.purs:13]:
```suggestion
-- | Results in `Nothing` if the string contains a non-hexadecimal number,
```
little error.

File: CHANGELOG.md [hunk 4]
```diff
@@ -38,6 +38,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - `Ctl.Examples.PlutusV2.OneShotMinting` example, demonstrating the use of `applyArgs` with a PlutusV2 script ([#1038](https://github.com/Plutonomicon/cardano-transaction-lib/pull/1038))
 - New `Contract.Test.Utils` assertions and checks: `assertOutputHasRefScript`, `checkOutputHasRefScript`, `checkTxHasMetadata` ([#1044](https://github.com/Plutonomicon/cardano-transaction-lib/pull/1044))
 - `Parallel` instance to `Contract` monad. Parallel capabilities are in the associated `ParContract` datatype ([#1037](https://github.com/Plutonomicon/cardano-transaction-lib/issues/1037))
+- `Contract.Cbor.hexToCborBytes` and `Contract.Cbor.hexToCbor` to decode a hexadecimal string to a `CborBytes` or `Cbor`, respectively. ([#850](https://github.com/Plutonomicon/cardano-transaction-lib/issues/850))
```
Comment (ngua) [inline CHANGELOG.md:41]: I'm not sure if there's a reason I'm not aware of, but is it worth it to export both of these? I think we could just export a function to work with `Cbor` and keep `CborBytes` for internal use. I think users will wonder what the difference is between the two types, then they'll need to read through the documentation/source more closely to figure it out. They can just use `unwrap` to get the `CborBytes` if that's really necessary

File: src/Contract/Cbor.purs [hunk 5]
```diff
@@ -0,0 +1,16 @@
+module Contract.Cbor
```
Comment (ngua) [inline src/Contract/Cbor.purs:1]: Please add module documentation

**Review by ngua — COMMENTED** [review]
It might also be nice to export a wrapper around `byteArrayToHex` just for completeness' sake (so `cborToHex`). That seems like a pretty common use case to me and it would save users from having to import another module and then `wrap` the result themselves

File: src/Contract/Cbor.purs (hunk 5 above)
Comment (MitchyCola) [reply src/Contract/Cbor.purs:2]: Added

File: CHANGELOG.md (hunk 4 above)
Comment (MitchyCola) [reply CHANGELOG.md]: I agree. It has been removed.

**Review by klntsky — CHANGES_REQUESTED** [review]
@ngua `Cbor` is our internal type, unlike `CborBytes`, isn't it? I don't get why do we need to expose it at all. `TextEnvelope` is the only place where it is used, and I think there's no reason to expose `TextEnvelope` at all (it is not exposed).

I think this PR should be made for CborBytes instead.

**Review by jy14898 — APPROVED** [review]
LGTM, please merge develop

File: src/Contract/CborBytes.purs [hunk 6]
```diff
@@ -0,0 +1,16 @@
+-- | A module with CBOR-related functionality.
+module Contract.CborBytes (module CborBytes) where
+
+import Ctl.Internal.Types.CborBytes
+  ( CborBytes(CborBytes)
+  , byteLength
```
Comment (klntsky) [inline src/Contract/CborBytes.purs:6]: Let's not export functions more than once from `Contract.*` modules. This one is in `Contract.Prim.ByteArray`.

**Review by klntsky — CHANGES_REQUESTED** [review]

File: src/Contract/CborBytes.purs (hunk 6 above)
Comment (MitchyCola) [reply src/Contract/CborBytes.purs:6]: It is not exactly the same, but it does use `Contract.Prim.ByteArray` inside the function: https://github.com/Plutonomicon/cardano-transaction-lib/blob/014e2b0119a16bd5ca936ac28d26ffeea8e9db35/src/Internal/Types/CborBytes.purs#L61-L62

We can be easily remove from the export list, as it is a simple function. Or maybe rename it to `cborByteLength`?

Comment (klntsky) [reply src/Contract/CborBytes.purs:6]: OK I didn't notice it's different

Comment (klntsky) [reply src/Contract/CborBytes.purs:6]: Yes, a rename would be good.

**Review by klntsky — APPROVED** [review]

**Pull request merged**